index.test.js

```diff
@@ -8,53 +8,183 @@ describe('[Exercise 1] trimProperties', () => {
     const actual = utils.trimProperties(input)
     expect(actual).toEqual(expected)
   })
-  // test('[2] returns a copy, leaving the original object intact', () => {})
+  test('[2] returns a copy, leaving the original object intact', () => {
+    const input = { foo: '  foo ', bar: 'bar ', baz: ' baz' }
+    const copy = utils.trimProperties(input)
+    expect(input).not.toEqual(copy)
+  })
+
 })
 
 describe('[Exercise 2] trimPropertiesMutation', () => {
-  // test('[3] returns an object with the properties trimmed', () => {})
-  // test('[4] the object returned is the exact same one we passed in', () => {})
+  test('[3] returns an object with the properties trimmed', () => {
+    const input = { foo: '  foo ', bar: 'bar ', baz: ' baz' }
+    const expected = { foo: 'foo', bar: 'bar', baz: 'baz' }
+    const actual = utils.trimProperties(input)
+    expect(actual).toEqual(expected)
+  })
+
+  test('[4] the object returned is the exact same one we passed in', () => {
+    const input = { foo: '  foo ', bar: 'bar ', baz: ' baz' }
+    const copy = utils.trimPropertiesMutation(input)
+    expect(copy).toEqual(input)
+    expect(copy).toBe(input)
+  }
+  )
 })
 
 describe('[Exercise 3] findLargestInteger', () => {
-  // test('[5] returns the largest number in an array of objects { integer: 2 }', () => {})
+  test('[5] returns the largest number in an array of objects { integer: 2 }', () => {
+    const array = [
+      {integer: 10}, 
+      {integer: 543}, 
+      {integer: 93}, 
+      {integer: 0}, 
+      {integer: -20}, 
+      {integer: 6}
+    ]
+    const result = utils.findLargestInteger(array)
+    expect(result).toBe(543)
+  })
 })
 
 describe('[Exercise 4] Counter', () => {
   let counter
   beforeEach(() => {
     counter = new utils.Counter(3) // each test must start with a fresh couter
   })
-  // test('[6] the FIRST CALL of counter.countDown returns the initial count', () => {})
-  // test('[7] the SECOND CALL of counter.countDown returns the initial count minus one', () => {})
-  // test('[8] the count eventually reaches zero but does not go below zero', () => {})
+
+  test('[6] the FIRST CALL of counter.countDown returns the initial count', () => {
+    const firstCount = counter.countDown()
+    expect(firstCount).toBe(3)
+  })
+
+  test('[7] the SECOND CALL of counter.countDown returns the initial count minus one', () => {
+    counter.countDown()
+    const secondCount = counter.countDown()
+    expect(secondCount).toBe(2)
+  })
+
+  test('[8] the count eventually reaches zero but does not go below zero', () => {
+    counter.countDown()
+    counter.countDown()
+    counter.countDown()
+    const fourthCount = counter.countDown()
+    const fifthCount = counter.countDown()
+
+    expect(fourthCount).toBe(0)
+    expect(fourthCount).not.toBeLessThan(0)
+    expect(fifthCount).not.toBeLessThan(0)
```

Comment:
Nit: Since you are checking if 4th count is 0 before less than 0 have your check for the 5th count be in the same order